From 9bf42cf63c765f8e4556af3ffdab796d0c733374 Mon Sep 17 00:00:00 2001
From: Byron McCollum <byron.mccollum@rackspace.com>
Date: Thu, 17 Dec 2015 09:12:42 -0600
Subject: [PATCH] Configurable options in deploy.sh for verify-maas playbook

This change introduces two new environment variables that control the
behavior of the verify-maas playbook: VERIFY_MAAS, and VERIFY_MAAS_DELAY.

VERIFY_MAAS (default 'yes') will control if the verify-maas playbook should
be run, and VERIFY_MAAS_DELAY (default '30') controls the sleep value before
the verify-maas playbook is run.

Fixes #621
---
 scripts/deploy.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/scripts/deploy.sh b/scripts/deploy.sh
index 26b8496..f539ba8 100755
--- a/scripts/deploy.sh
+++ b/scripts/deploy.sh
@@ -9,6 +9,8 @@ export DEPLOY_HAPROXY=${DEPLOY_HAPROXY:-"no"}
 export DEPLOY_OA=${DEPLOY_OA:-"yes"}
 export DEPLOY_ELK=${DEPLOY_ELK:-"yes"}
 export DEPLOY_MAAS=${DEPLOY_MAAS:-"no"}
+export VERIFY_MAAS=${VERIFY_MAAS:-"yes"}
+export VERIFY_MAAS_DELAY=${VERIFY_MAAS_DELAY:-"30"}
 export DEPLOY_TEMPEST=${DEPLOY_TEMPEST:-"no"}
 export DEPLOY_CEILOMETER=${DEPLOY_CEILOMETER:-"no"}
 export DEPLOY_CEPH=${DEPLOY_CEPH:-"no"}
@@ -142,8 +144,10 @@ run_ansible horizon_extensions.yml
 # deploy and configure RAX MaaS
 if [[ "${DEPLOY_MAAS}" == "yes" ]]; then
   run_ansible setup-maas.yml
-  sleep 30
-  run_ansible verify-maas.yml
+  if [[ "${VERIFY_MAAS}" == "yes" ]]; then
+      sleep $VERIFY_MAAS_DELAY
+      run_ansible verify-maas.yml
+  fi
 fi
 
 # deploy and configure the ELK stack
-- 
2.6.0

